File: app/actors/SocketActor.java
package actors;

import akka.actor.*;

public class SocketActor extends UntypedActor {

    public static Props props(ActorRef out) {
        return Props.create(SocketActor.class, out);
    }

    private final ActorRef out;

    public SocketActor(ActorRef out) {
        this.out = out;
    }

    public void onReceive(Object message) throws Exception {
        if (message instanceof String) {
            out.tell("I received your message: " + message, self());
        }
    }
}

File: app/actors/SocketActor.scala
package actors

import java.lang.IllegalArgumentException

import akka.actor.{ Actor, ActorRef, Props }
import de.htwg.se.nmm.Game
import models.{ GameObserver, User }
import play.api.libs.json.Json
import services.{ HTTPWorker, JsonWorker }

/**
 * Created by funkemarkus on 16.12.16.
 */
object SocketActor {
  def props(out: ActorRef) = Props(new SocketActor(out))
}

class SocketActor(out: ActorRef) extends Actor {
  val httpWorker: HTTPWorker = new HTTPWorker(this);

  def receive = {
    case msg: String =>
      //out ! ("Hi, I received your message: " + msg)
      try {
        new JsonWorker().processJson(httpWorker, msg);
      } catch {
        case ex: Exception => {
          val message = Json.obj("code" -> "400", "message" -> "Bad Request (see logs for more details)")
          ex.printStackTrace()
          out ! message.toString()
        }
      }
  }

  def sendMessage(data: String) = {
    //println(data)
    out ! data
  }
}
